src/index.ts

```diff
@@ -142,35 +142,36 @@ export function either<T>(parser: Parser<T>): Parser<{isError: boolean, value: T
   });
 };
 
-// coroutine :: (() -> Iterator (Parser e a s)) -> Parser e a s
-export function coroutine<T>(g: FnReturingParserIterator<T>): Parser<T> {
+// coroutine :: (() -> e) -> Parser e a s
```

Comment:
You can remove this line. Since moving to typescript, there isn't a lot of value in these signatures anyway.

Reply:
I will remove all signatures in next typings refactor